### PR TITLE
Initial backend

### PR DESCRIPTION
**Pull request by CristobalMe**

### CODE REVIEW

File: frontend/src/components/spinningBanner/imagesSpinningBaner/a
Comment (toriqyanM) [inline frontend/src/components/spinningBanner/imagesSpinningBaner/a:1]: Remove this file

File: backend/server.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  })
+);
+sessionStore.sync();
+
+const eventRoutes = require("./routes/eventRoutes")
+const userRoutes = require("./routes/userRoutes")
+
+app.get('/', (req, res) => {
+    res.send('Welcome to my app!')
+})
```
Comment (toriqyanM) [inline backend/server.js:48]: Remove

File: backend/server.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+const newSequelize = new Sequelize('event_finder', 'postgres', '1025323', {
+    host: 'localhost',
+    dialect: 'postgres'
+});
+
+const SequelizeStore = SequelizeStoreInit(session.Store);
+const sessionStore = new SequelizeStore({
+  db: newSequelize
+});
```
Comment (toriqyanM) [inline backend/server.js:25]: Sequelize is an equivalence to prisma. I don't think you are using it for your server though.

File: backend/server.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+newSequelize.sync({ alter: true })
+  .then(() => {
+    const port = 3000;
+    app.listen(port, () => {
+      console.log(`App is listening on port ${port}`);
+    });
+  })
+  .catch(error => {
+    console.error('Unable to connect to the database:', error);
+  });
```
Comment (toriqyanM) [inline backend/server.js:62]: Same comment on sequelize

File: backend/routes/userRoutes.js
Comment (toriqyanM) [inline backend/routes/userRoutes.js:1]: Again code style: remove empty lines

File: backend/routes/userRoutes.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  req.session.user = newUser;
+
+  res.json(newUser)
+})
+
+router.put('/:id', async(req, res) => {
+  const { id } = req.params
+  const {  username, email, password  } = req.body
+
+  const updatedUser = await prisma.user.update({
```
Comment (toriqyanM) [inline backend/routes/userRoutes.js:40]: What should we expect if the user id doesn't exist in db? Like a malicious api call to your server?

File: backend/routes/userRoutes.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+})
+
+
+
+router.post('/login', async (req, res) => {
+    const { username, password } = req.body;
+
+    try {
+      // Find the user by username
+      const User = await prisma.user.findFirst({ where: { username: username } });
```
Comment (toriqyanM) [inline backend/routes/userRoutes.js:70]: Again code style: variable name is lower capitalized.

File: backend/.gitignore [hunk 6]
```diff
@@ -0,0 +1,4 @@
+node_modules
+**/node_modules
+# Keep environment variables out of version control
```
Comment (toriqyanM) [inline backend/.gitignore:3]: Add .DS_Store to gitignore as well

**Review by toriqyanM — CHANGES_REQUESTED** [review]

File: backend/routes/userRoutes.js (hunk 4 above)
Comment (yijial) [reply backend/routes/userRoutes.js:40]: do we also need to check if the username is the same as the one logged in?

File: backend/routes/userRoutes.js [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+  const deletedUser = await prisma.user.delete({
+    where: { id: parseInt(id) }
+  })
+
+  res.json(deletedUser)
+
+})
+
+
```
Comment (yijial) [inline backend/routes/userRoutes.js:63]: let's play the game: who can find extra empty lines?

**Review by yijial — COMMENTED** [review]
Thanks for breaking down the changes. This is a lot more readable

File: backend/routes/userRoutes.js (hunk 5 above)
Comment (CristobalMe) [reply backend/routes/userRoutes.js:70]: Done

File: backend/.gitignore (hunk 6 above)
Comment (CristobalMe) [reply backend/.gitignore:3]: Done

File: backend/server.js (hunk 1 above)
Comment (CristobalMe) [reply backend/server.js:48]: Done

File: frontend/src/components/spinningBanner/imagesSpinningBaner/a
Comment (CristobalMe) [reply frontend/src/components/spinningBanner/imagesSpinningBaner/a:1]: Done

File: backend/server.js (hunk 2 above)
Comment (CristobalMe) [reply backend/server.js:25]: To do

File: backend/server.js (hunk 3 above)
Comment (CristobalMe) [reply backend/server.js:62]: To do

File: backend/routes/userRoutes.js (hunk 4 above)
Comment (CristobalMe) [reply backend/routes/userRoutes.js:40]: Can I check it in the frontend?

Comment (CristobalMe) [reply backend/routes/userRoutes.js:40]: > What should we expect if the user id doesn't exist in db? Like a malicious api call to your server?

I think I'm going to delete all the actions related to user (except get and delete)

Comment (toriqyanM) [reply backend/routes/userRoutes.js:42]: I don't think I'm following your decision here in deleting all the actions. Can you elaborate a bit more?
I am suggesting for some sanity checks and error handling for update's use case here.

Comment (toriqyanM) [reply backend/routes/userRoutes.js:42]: Why not check username right here? You have username passed in to the req.body, and updatedUser.username retrieved from prisma table

File: backend/routes/userRoutes.js
Comment (CristobalMe) [reply backend/routes/userRoutes.js:1]: Done

File: backend/routes/userRoutes.js (hunk 4 above)
Comment (CristobalMe) [reply backend/routes/userRoutes.js:42]: To Do:
check username in userRoutes

File: backend/routes/userRoutes.js (hunk 7 above)
Comment (CristobalMe) [reply backend/routes/userRoutes.js]: Done

**Pull request merged**